### PR TITLE
Add support for connecting to postgresql database

### PR DESCRIPTION
**Pull request by SquareMike**
@pierre Please review and merge.  If this gets merged I will submit a PR to killbill-admin-ui which makes the ddl compatible with postgresql.

### CODE REVIEW

**Comment by SquareMike** [comment]
@pierre Do I need to take action based on the failed Travis build?  The error doesn't seem related to the change I made but I am more than happy to investigate if it's a real issue.

```mime-types-data-3.2016.0521 requires ruby version >= 2.0, which is incompatible```

File: Gemfile [hunk 1]
```diff
@@ -55,6 +55,7 @@ if defined?(JRUBY_VERSION)
 
   gem 'activerecord-jdbc-adapter', '~> 1.3.9'
   gem 'activerecord-jdbcmysql-adapter', '~> 1.3.9'
+  gem 'activerecord-jdbcpostgresql-adapter', '~> 1.3.9'
```
Comment (pierre) [inline Gemfile]: We should also add `gem 'pg'` in the `else` block below, for Ruby MRI (especially useful during development, where restarts are much faster than with JRuby).

**Comment by pierre** [comment]
> Do I need to take action based on the failed Travis build? The error doesn't seem related to the change I made but I am more than happy to investigate if it's a real issue.

Yeah, it looks unrelated indeed. I actually didn't realize the build was constantly failing recently. The Travis environment is quite flaky, so errors are often expected unfortunately. We usually deep dive into the build logs of the various repos periodically, to verify failures are benign. It looks like this one fell through the cracks.

https://github.com/killbill/killbill-admin-ui/issues/45 for tracking. If you have time, feel free to take a look at it, otherwise, we'll add it to our (long) TODO list! 😼

File: Gemfile (hunk 1 above)
Comment (SquareMike) [inline Gemfile]: Cool, added, thanks.

**Comment by pierre** [comment]
Thanks!

**Comment by SquareMike** [comment]
Out of curiousity, does this automatically get pushed out for distribution or does a process need to occur?  When I start my kaui container I get:

```
fenix-admin_1  | I, [2016-12-29T18:22:30.659000 #31]  INFO -- :       Starting download of org.kill-bill.billing.kaui:kaui-standalone:war:0.8.0 to /tmp/d20161229-31-1tpady3
```

**Comment by pierre** [comment]
> Out of curiousity, does this automatically get pushed out for distribution or does a process need to occur?

Releases are done manually - we are currently actively working on a few new screens and features in Kaui, so I expect a new build to be out in the next couple of weeks, along with Kill Bill 0.18.2.

FYI, KPM pulls artifacts from [Maven Central](https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22org.kill-bill.billing.kaui%22%20AND%20a%3A%22kaui-standalone%22) and can be configured through a [YAML file](https://github.com/killbill/killbill-cloud/blob/master/kpm/install_example.yml#L24). You could point it to your own Nexus instance if you have one (where you could push custom builds). Or alternatively, you can simply `docker cp` a custom build for now and restart the container.

**Comment by SquareMike** [comment]
Got it, thanks.

**Comment by rmgas1** [comment]
Hi @pierre and @SquareMike,

Not sure if this is the right forum for these questions, if not, please point me to where I should ask them.

Tracing through [Kill Bill users mailing-list › PostgreSQL support](https://groups.google.com/forum/#!searchin/killbilling-users/postgres%7Csort:relevance/killbilling-users/ArdWn_iE4SU/TPXbzxiPFgAJ) I read that PostgreSQL support in Kaui was tracked in issue #9. Issue #9 is closed with a reference to a fix in issue #10 (this one).

The [killbill-admin-ui ddl.sql](https://raw.githubusercontent.com/killbill/killbill-admin-ui/master/db/ddl.sql) fails on PostgreSQL (having a look at it, it is still using MySQL) but looking at @SquareMike's first comment I gather the change to make the DDL compatible with PostgreSQL is probably done, but not yet public.

So two questions:

1. When do you think you will make the PostgreSQL DDL change public?
2. Is there anything else that needs to be updated to make Kaui work with PostgreSQL?

I'm setting up Killbill for a production deployment, and the preference is to use PostgreSQL as the DB for both, so your response would be much appreciated.

**Comment by pierre** [comment]
> Not sure if this is the right forum for these questions, if not, please point me to where I should ask them.

Best is usually to reach out the [mailing-list](https://groups.google.com/forum/#!forum/killbilling-users).

> The [killbill-admin-ui ddl.sql](https://raw.githubusercontent.com/killbill/killbill-admin-ui/master/db/ddl.sql) fails on PostgreSQL (having a look at it, it is still using MySQL) but looking at @SquareMike's first comment I gather the change to make the DDL compatible with PostgreSQL is probably done, but not yet public.

Indeed, while the driver was added, the DDL was never updated. I've just pushed https://github.com/killbill/killbill-admin-ui/commit/5ae476a0485dedacd046b261fc266509228cd3f8 (untested -- let me know if it works).

Note that it assumes that the [Kill Bill DDL bridge](https://github.com/killbill/killbill/blob/master/util/src/main/resources/org/killbill/billing/util/ddl-postgresql.sql) is installed.

> Is there anything else that needs to be updated to make Kaui work with PostgreSQL?

I don't think so, but if anything else isn't working as expected, let us know.

💭 We could also consider enabling PostgreSQL tests [on Travis](https://github.com/killbill/killbill-admin-ui/blob/master/.travis.yml).

**Pull request merged**